swift/ql/lib/codeql/swift/elements/decl/MethodDecl.qll

```diff
@@ -21,4 +21,33 @@ class MethodDecl extends AbstractFunctionDecl {
     or
     this = getAMember(any(ProtocolDecl c))
   }
+
+  /**
+   * Holds if this function is called `funcName` and its a member of a
+   * class, struct, extension, enum or protocol call `typeName`.
+   */
+  predicate hasQualifiedName(string typeName, string funcName) {
```

Comment:
This should probably be `cached`.

Reply:
Done.  I wasn't sure whether to do the three argument version as its only a small bit of logic + a call to the two argument version.  So I left it uncached for now.

Reply:
Yeah, I don't think the three-argument version should be cached (because, as you say, it's only a small wrapper around cached stuff at this point).